File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/SelfSigner.kt
package com.example.kotlinmpplib

import breez_sdk_liquid.*
class SelfSigner {
   // ANCHOR: self-signer
    fun connectWithSelfSigner(signer: Signer) {               

        // Create the default config, providing your Breez API key
        val config : Config = defaultConfig(LiquidNetwork.MAINNET, "<your Breez API key>")

        // Customize the config object according to your needs
        config.workingDir = "path to an existing directory"

        try {
            val connectRequest = ConnectWithSignerRequest(config)
            val sdk = connectWithSigner(connectRequest, signer)
        } catch (e: Exception) {
            // handle error
        }
    }
     // ANCHOR_END: self-signer
  }

File: snippets/swift/BreezSDKExamples/Sources/SelfSigner.swift
import BreezSDKLiquid

  // ANCHOR: self-signer
func connectWithSelfSigner(signer: Signer) throws -> BindingLiquidSdk? {
  
    let mnemonic = "<mnemonic words>"

    // Create the default config, providing your Breez API key
    var config = try defaultConfig(network: LiquidNetwork.mainnet, breezApiKey: "<your-Breez-API-key>")

    // Customize the config object according to your needs
    config.workingDir = "path to an existing directory"

    let connectRequest = ConnectWithSignerRequest(config: config)
    let sdk = try? connectWithSigner(req: connectRequest, signer: signer)

    return sdk
}
// ANCHOR: self-signer
